### PR TITLE
chore_: enable nightly codeclimate reports

### PR DESCRIPTION
**Pull request by igor-sirotin**
Continue of https://github.com/status-im/status-go/pull/5101

### CODE REVIEW

**Review by jakubgs — APPROVED** [review]
Probably fine?

**Review by siddarthkay — APPROVED** [review]

**Comment by yakimant** [comment]
Wonder, how it will work on failing tests.
Nightly tests always fail. Will it exit the script before submitting to codeclimate?

Also - nightly runs are checking for instability. Running many times to catch failures.
Don't we need the opposite - super stable master run to submit the coverage? (eg retrying failed tests to overcome flakiness)

**Comment by igor-sirotin** [comment]
> Wonder, how it will work on failing tests.

TLDR: I thought of this, it potentially could be a problem, but was hard to predict, decided to give it a try.

-----

Doing it right, coverage should be run on `master` after each merged PR. So that the next PR is compared to the latest calculated coverage.

Since we don't have tests running on master, and tests are taking ~40min, I looked on the nightly. 

The rerun and failing should in theory only help. We save each run separately. After all tests finished we sum up the reports and post to codeclimate.

It will eventually submit to codeclimate no matter of results.

-----

So, first of all, I'd just give it a try and see how it works.

Eventually we can change the behaviour after we:
- make the tests not flaky
- make the tests run faster (parallelize)

Then we can run tests in each master.

cc @yakimant @jakubgs @siddarthkay

**Comment by yakimant** [comment]
Nice plan!

But can you please check the logic before merging?
Fo me it looks like failing tests will not allow coverage publishing:
https://github.com/status-im/status-go/blob/068ae3b67cdb00e383f9b87601c32680dfa62b1f/_assets/scripts/run_unit_tests.sh#L130
